lib/groonga-query-log/command/check-performance-regression.rb

```diff
@@ -0,0 +1,389 @@
+
+          parser.on("--slow-operation-ratio=PERCENTAGE",
+                    Float,
+                    "Use PERCENTAGE% as threshold to detect slow operations.",
+                    "Example: --slow-operation-ratio=#{@options[:slow_operation_ratio]} means",
+                    "changed amount of operation time is #{@options[:slow_operation_ratio]}% or more.",
+                    "(#{@options[:slow_operation_ratio]})") do |ratio|
+            @options[:slow_operation_ratio] = ratio
+          end
+
+          parser.on("--slow-response-ratio=PERCENTAGE",
+                    Float,
+                    "Use PERCENTAGE% as threshold to detect slow responses.",
+                    "Example: --slow-response-ratio=#{@options[:slow_response_ratio]} means",
+                    "changed amount of response time is #{@options[:slow_response_ratio]}% or more.",
+                    "(#{@options[:slow_response_ratio]})") do |ratio|
+            @options[:slow_response_ratio] = ratio
+          end
+
+          parser.on("--slow-operation-threshold=THRESHOLD",
+                    Float,
+                    "Use THRESHOLD seconds to detect slow operations.",
+                    "(#{@options[:slow_operation_threshold]})") do |threshold|
+            @options[:slow_operation_threshold] = threshold
+          end
+
+          parser.on("--slow-response-threshold=THRESHOLD",
+                    Float,
+                    "Use THRESHOLD seconds to detect slow responses.",
+                    "(#{@options[:slow_response_threshold]})") do |threshold|
+            @options[:slow_response_threshold] = threshold
+          end
+        end
+      end
+
+      def group_statistics(old_statistics, new_statistics)
+        old_queries = {}
+        new_queries = {}
+        old_statistics.count.times do |i|
+          next if i > new_statistics.count - 1
+          if cached_query?(old_statistics[i])
+            @n_cached_query += 1
+            next
+          end
+          next if different_query?(old_statistics[i], new_statistics[i])
+
+          raw_command = old_statistics[i].raw_command
+          next if not filter_query?(raw_command)
+
+          if old_queries[raw_command]
+            statistics = old_queries[raw_command]
+            statistics << old_statistics[i]
+            old_queries[raw_command] = statistics
+          else
+            old_queries[raw_command] = [old_statistics[i]]
+          end
+
+          if new_queries[raw_command]
+            statistics = new_queries[raw_command]
+            statistics << new_statistics[i]
+            new_queries[raw_command] = statistics
+          else
+            new_queries[raw_command] = [new_statistics[i]]
+          end
+        end
+        [old_queries, new_queries]
+      end
+
+      def filter_query?(query)
+        if @options[:input_filter_query]
+          if @options[:input_filter_query].include?(query)
+            true
+          else
+            false
+          end
```

Comment:
Only `@options[:input_filter_query].include?(query)` is enough. `if` is needless.

Reply:
fixed.